library/spdm_common_lib/libspdm_com_opaque_data.c

```diff
@@ -233,64 +239,98 @@ libspdm_process_opaque_data_supported_version_data(libspdm_context_t *spdm_conte
 
     if (libspdm_get_connection_version (spdm_context) >= SPDM_MESSAGE_VERSION_12) {
         spdm_general_opaque_data_table_header = data_in;
-        if (spdm_general_opaque_data_table_header->total_elements != 1) {
+        if (spdm_general_opaque_data_table_header->total_elements < 1) {
             return LIBSPDM_STATUS_INVALID_MSG_FIELD;
         }
         opaque_element_table_header =
             (const void *)(spdm_general_opaque_data_table_header + 1);
+
+        element_num = spdm_general_opaque_data_table_header->total_elements;
     } else {
         general_opaque_data_table_header = data_in;
         if ((general_opaque_data_table_header->spec_id !=
              SECURED_MESSAGE_OPAQUE_DATA_SPEC_ID) ||
             (general_opaque_data_table_header->opaque_version !=
              SECURED_MESSAGE_OPAQUE_VERSION) ||
-            (general_opaque_data_table_header->total_elements != 1)) {
+            (general_opaque_data_table_header->total_elements < 1)) {
             return LIBSPDM_STATUS_INVALID_MSG_FIELD;
         }
         opaque_element_table_header =
             (const void *)(general_opaque_data_table_header + 1);
-    }
 
-    opaque_element_support_version =
-        (const void *)(opaque_element_table_header + 1);
-    if ((opaque_element_support_version->sm_data_version !=
-         SECURED_MESSAGE_OPAQUE_ELEMENT_SMDATA_DATA_VERSION) ||
-        (opaque_element_support_version->sm_data_id !=
-         SECURED_MESSAGE_OPAQUE_ELEMENT_SMDATA_ID_SUPPORTED_VERSION) ||
-        (opaque_element_support_version->version_count == 0)) {
-        return LIBSPDM_STATUS_INVALID_MSG_FIELD;
+        element_num = general_opaque_data_table_header->total_elements;
     }
 
-    version_count = opaque_element_support_version->version_count;
+    for (element_index = 0; element_index < element_num;
+         element_index++) {
 
-    if ((opaque_element_table_header->id != SPDM_REGISTRY_ID_DMTF) ||
-        (opaque_element_table_header->vendor_len != 0) ||
-        (opaque_element_table_header->opaque_element_data_len !=
-         sizeof(secured_message_opaque_element_supported_version_t) +
-         sizeof(spdm_version_number_t) * version_count)) {
-        return LIBSPDM_STATUS_INVALID_MSG_FIELD;
-    }
+        /*check element header id*/
+        if ((opaque_element_table_header->id < SPDM_REGISTRY_ID_DMTF) ||
```

Comment:
SPDM_REGISTRY_ID_DMTF is 0.

How could `opaque_element_table_header->id < SPDM_REGISTRY_ID_DMTF` happen?

Reply:
I have deleted the code.